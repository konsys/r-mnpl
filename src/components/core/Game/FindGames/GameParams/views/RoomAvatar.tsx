import React from "react";

export default function RoomAvatar() {
  return <>RoomAvatar</>;
}
