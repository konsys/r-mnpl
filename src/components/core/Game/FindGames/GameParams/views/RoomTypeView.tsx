import React from "react";
import { RoomType } from "../../CreateGameModal";
import { useTranslation } from "react-i18next";

export default function RoomTypeView({ type }: { type: RoomType }) {
  const { t } = useTranslation();
  return (
    <>
      {
        (type === RoomType.QUICK && t(RoomType.QUICK),
        type === RoomType.REGULAR && t(RoomType.REGULAR),
        type === RoomType.RETRO && t(RoomType.RETRO),
        type === RoomType.SHUFFLE && t(RoomType.SHUFFLE),
        type === RoomType.ROULETTE && t(RoomType.ROULETTE))
      }
    </>
  );
}
