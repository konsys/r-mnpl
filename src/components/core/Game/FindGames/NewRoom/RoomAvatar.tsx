import { Params } from "config/params";
import React from "react";

export default function RoomAvatar({
  avatar,
  name,
}: {
  avatar: string;
  name: string;
}) {
  return (
    <>
      <img src={`${Params.BASE_URL}/${avatar}`} alt={name} />
    </>
  );
}
