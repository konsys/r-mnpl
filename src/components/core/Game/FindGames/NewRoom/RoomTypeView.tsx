import { Grid, Typography } from "@material-ui/core";
import {
  IRoomState,
  RoomPortalFieldType,
  RoomType,
  RoomTypeName,
} from "stores/Game/Room/NewRoomStore";

import Autorenew from "@material-ui/icons/Autorenew";
import CasinoIcon from "@material-ui/icons/Casino";
import EvStationIcon from "@material-ui/icons/EvStation";
import LocationSearchingIcon from "@material-ui/icons/LocationSearching";
import RadioIcon from "@material-ui/icons/Radio";
import React from "react";
import RestoreIcon from "@material-ui/icons/Restore";
import SendIcon from "@material-ui/icons/Send";
import ShuffleIcon from "@material-ui/icons/Shuffle";
import VideogameAssetIcon from "@material-ui/icons/VideogameAsset";
import VisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import { useTranslation } from "react-i18next";

export default function RoomTypeView({ room }: { room: IRoomState }) {
  const { t } = useTranslation();
  let name = "";
  let className = "";
  let icon = <VideogameAssetIcon />;

  switch (room.roomType) {
    case RoomType.REGULAR:
      name = RoomTypeName.REGULAR;
      className = RoomType.REGULAR;

      break;
    case RoomType.QUICK:
      name = RoomTypeName.QUICK;
      className = RoomType.QUICK;
      icon = <SendIcon />;
      break;
    case RoomType.RETRO:
      name = RoomTypeName.RETRO;
      className = RoomType.RETRO;
      icon = <RadioIcon />;
      break;
    case RoomType.ROULETTE:
      name = RoomTypeName.ROULETTE;
      className = RoomType.ROULETTE;
      icon = <LocationSearchingIcon />;
      break;
    case RoomType.SHUFFLE:
      name = RoomTypeName.SHUFFLE;
      className = RoomType.SHUFFLE;
      icon = <ShuffleIcon />;
      break;
  }

  return (
    <Grid
      container
      justify="flex-start"
      alignContent="center"
      alignItems="center"
      spacing={1}
    >
      <Grid item className={className}>
        {icon}
      </Grid>
      <Grid item className={className}>
        {t(name)}
      </Grid>
      <Grid item>
        {room.restarts && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <RestoreIcon />
              {t("With restarts")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.autostart && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <Autorenew />
              {t("Game autostart")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.privateRoom && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignContent="center">
              <VisibilityOffIcon />
              {t("Private room")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.portalType === RoomPortalFieldType.PORTAL && (
          <Typography color="textSecondary" variant="subtitle2">
            {t("With portal")}
          </Typography>
        )}
        {room.portalType === RoomPortalFieldType.ROULETTE && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <CasinoIcon />
              {t("Roulette")}
            </Grid>
          </Typography>
        )}
        {room.portalType === RoomPortalFieldType.RUSSIAN_ROULETTE && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <EvStationIcon />
              {t("Russian roulette")}
            </Grid>
          </Typography>
        )}
      </Grid>
    </Grid>
  );
}
