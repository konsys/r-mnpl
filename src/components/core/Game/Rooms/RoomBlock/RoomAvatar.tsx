import { Grid, Typography } from "@material-ui/core";

import AddIcon from "@material-ui/icons/Add";
import { AccessAlarm } from "@material-ui/icons";
import HighlightOffIcon from "@material-ui/icons/HighlightOff";
import React from "react";
import { createImgPath } from "utils/fields.utils";
import { useTranslation } from "react-i18next";

export default function RoomAvatar({
  avatar,
  name,
  addPlayer,
  removePlayer,
  roomId,
  isMe,
  isMyRoom,
}: {
  avatar?: string;
  name?: string;
  addPlayer: () => void;
  removePlayer: ((id: string) => void) | null;
  roomId: string;
  isMe: boolean;
  isMyRoom: boolean;
}) {
  const { t } = useTranslation();

  return (
    <>
      <Grid container justify="center" alignItems="center" direction="column">
        <Grid item className="avatar">
          <Grid
            container
            justify="center"
            alignItems="center"
            alignContent="center"
            direction="column"
          >
            <Grid item>
              {avatar && name ? (
                <>
                  <div className="avatarWrap">
                    {isMe && removePlayer && (
                      <HighlightOffIcon onClick={() => removePlayer(roomId)} />
                    )}
                    <img src={createImgPath(avatar)} alt={name} />
                  </div>
                </>
              ) : (
                <Grid
                  container
                  justify="center"
                  alignItems="center"
                  alignContent="center"
                  direction="column"
                >
                  <Grid item className="addPlayerWrap">
                    {!isMyRoom ? (
                      <AddIcon onClick={addPlayer} />
                    ) : (
                      <span className="noCursor">
                        <AccessAlarm />
                      </span>
                    )}
                  </Grid>
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          {/* TODO add link to profile */}
          <Typography variant="subtitle2">
            {isMyRoom ? name : t("Enter room")}
          </Typography>
        </Grid>
      </Grid>
    </>
  );
}
