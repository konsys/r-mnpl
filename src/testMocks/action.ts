import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
