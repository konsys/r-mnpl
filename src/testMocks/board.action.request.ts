import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}