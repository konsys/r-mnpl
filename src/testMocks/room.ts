import {
  IRoomResponce,
  IRoom,
  PlayerRoomStatus,
  RoomPortalFieldType,
  RoomStatus,
  RoomType,
} from "stores/Game/Rooms/RoomsModel";
import { testPlayer1, testPlayer2 } from "./user";

export const testPlayingRoom: IRoom = {
  roomId: "testRoomIdtestPlayingRoom",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PLAYING,
};

export const test4Room: IRoom = {
  roomId: "testRoomIdtest4Room",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 4,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PLAYING,
};

export const testPendingRoom1: IRoom = {
  roomId: "testRoomIdtestPendingRoom1",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PENDING,
};

export const testPendingRoom2: IRoom = {
  roomId: "testRoomIdtestPendingRoom2",
  creatorId: 2,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PENDING,
};

export const testRoomsResponce: IRoomResponce = {
  playersInRooms: 2,
  rooms: [testPlayingRoom],
};

export const testPendingRoomsResponce: IRoomResponce = {
  playersInRooms: 2,
  rooms: [testPendingRoom1, testPendingRoom2],
};
