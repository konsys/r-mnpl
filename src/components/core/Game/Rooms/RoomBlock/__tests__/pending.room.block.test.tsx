import { test4Room, testRoom } from "testMocks/room";

import PendingRoomBlock from "../PendingRoomBlock";
import React from "react";
import RoomAvatar from "../RoomAvatar";
import RoomTypeView from "../RoomTypeView";
import { shallow } from "enzyme";

describe("Room block test", () => {
  it("should render", () => {
    expect(
      shallow(<PendingRoomBlock room={testRoom} iHaveRoom={true} userId={1} />)
    ).toMatchSnapshot();
  });

  it("should render RoomTypeView", () => {
    expect(
      shallow(
        <PendingRoomBlock room={testRoom} iHaveRoom={true} userId={1} />
      ).find(RoomTypeView)
    ).toHaveLength(1);
  });

  it("should render RoomTypeViewp imgs", () => {
    expect(
      shallow(
        <PendingRoomBlock room={testRoom} iHaveRoom={true} userId={1} />
      ).find(RoomAvatar)
    ).toHaveLength(testRoom.playersNumber);
  });

  it("should render RoomTypeView props", () => {
    const wrap = shallow(
      <PendingRoomBlock room={test4Room} iHaveRoom={true} userId={1} />
    )
      .find(RoomAvatar)
      .get(0).props;

    expect(wrap.avatar).toStrictEqual(test4Room.players[0]?.avatar);
    expect(wrap.name).toStrictEqual(test4Room.players[0]?.name);
    expect(wrap.roomId).toStrictEqual(test4Room.roomId);

    const wrap2 = shallow(
      <PendingRoomBlock room={test4Room} iHaveRoom={true} userId={1} />
    )
      .find(RoomAvatar)
      .get(2).props;

    expect(wrap2.avatar).toStrictEqual("");
    expect(wrap2.name).toStrictEqual("");
  });

  it("should render RoomTypeView removePlayer func", () => {
    const wrap = shallow(
      <PendingRoomBlock room={test4Room} iHaveRoom={true} userId={undefined} />
    )
      .find(RoomAvatar)
      .get(0).props;

    expect(wrap.removePlayer).toStrictEqual(null);

    const wrap1 = shallow(
      <PendingRoomBlock room={test4Room} iHaveRoom={true} userId={1} />
    )
      .find(RoomAvatar)
      .get(0).props;

    expect(typeof wrap1.removePlayer).toBe("function");
  });
});
