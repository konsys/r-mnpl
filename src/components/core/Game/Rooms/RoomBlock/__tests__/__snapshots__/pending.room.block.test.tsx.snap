// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room block test should render 1`] = `ShallowWrapper {}`;
